google-cloud-spanner/src/main/java/com/google/cloud/spanner/connection/ClientSideStatementValueConverters.java

```diff
@@ -494,4 +495,46 @@ public String convert(String value) {
       return value.substring(7).trim();
     }
   }
+
+  /** Converter for converting Base64 encoded string to byte[] */
+  static class ProtoDescriptorsConverter implements ClientSideStatementValueConverter<byte[]> {
+
+    public ProtoDescriptorsConverter(String allowedValues) {}
+
+    @Override
+    public Class<byte[]> getParameterClass() {
+      return byte[].class;
+    }
+
+    @Override
+    public byte[] convert(String value) {
+      if (value.length() == 0 || value.equalsIgnoreCase("null")) {
+        return null;
+      }
+      try {
+        return Base64.getDecoder().decode(value);
+      } catch (IllegalArgumentException e) {
+        return null;
+      }
+    }
+  }
+
+  /** Converter for converting String that take in file path as input to String */
+  static class ProtoDescriptorsFileConverter implements ClientSideStatementValueConverter<String> {
```

Comment:
We supporting reading from the file here? If Yes where should file be located just on resources or local path ? 
We had similar discussion on https://github.com/googleapis/java-spanner/pull/2277#discussion_r1111874247.

Reply:
Yes we are supporting reading from a file path given through a client side statement. This is needed to support ORM's and other frameworks. This discussion is available in the design doc.
Just a small brief, in [PR#2277](https://github.com/googleapis/java-spanner/pull/2277#discussion_r1111874247), we were treating file as a resource which brings in complications as they rely on class path. However, in this method we instead used File system to read from relative or absolute path.